torch/distributed/tensor/parallel/fsdp.py

```diff
@@ -220,6 +221,66 @@ def _chunk_tensor(
         )
 
 
+def _chunk_dtensor(
```

Comment:
Can we reuse `distribute_tensor` API?

Reply:
Do you mean the DTensor.from_local().redistribute() part? Since we already know the layout here as well, so for now, it's better to use DTensor.from_local().redistribute().